functions/src/app.ts

```diff
@@ -0,0 +1,4 @@
+import * as admin from 'firebase-admin';
+
+// Do this only once at the top level.
+export const app = admin.initializeApp();
```

Comment:
Added this to make sure we init `app` only once and can re-use everywhere. Running the init in multiple files creates an error on deploy.

Reply:
But that's why I initialized different instances with different names for each instance.
I am not sure about deployment, but the error during runtime wouldn't exist with different names.
I wanted to keep each module independent as much as possible

Reply:
I don't think that's necessary -- different modules with different instances.

Reply:
Each module runs in a different instance regardless as they all run in different cloud functions.

Reply:
okay, that works too. but should we also make this change for the cloud function files for each model as right now we are initializing the admin SDK in every file.

we could implement it like keep the initialization in one single file that exports the initialized Admin SDK instance, just as it is currently being done in the web-client app.

Reply:
do you want me to make this change @sabind ?

Reply:
I think I made the changes everywhere necessary. I will revisit this one we merge this and the testing PR in.